microsoft-edge/webview2/concepts/frames.md

```diff
@@ -0,0 +1,498 @@
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2NavigationStartingEventArgs.AdditionalAllowedFrameAncestors Property](https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2navigationstartingeventargs)
+
+##### [Win32/C++](#tab/win32cpp)
+
+[ICoreWebView2NavigationStartingEventArgs2::put_AdditionalAllowedFrameAncestors property (get](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2navigationstartingeventargs2#get_additionalallowedframeancestors), [put)](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2navigationstartingeventargs2#put_additionalallowedframeancestors)
+
+---
+
+
+<!-- ====================================================================== -->
+## Example of using iframes in a host app
+
+This sample code shows how to use frame APIs, including:
+* `FrameCreated`
+   * `CoreWebView2FrameCreatedEventArgs`
+* `DOMContentLoaded`
+   * `CoreWebView2DOMContentLoadedEventArgs`
+* `ExecuteScript`
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+This sample code is condensed from [MainWindow.xaml.cs](https://github.com/MicrosoftEdge/WebView2Samples/blob/main/SampleApps/WebView2WpfBrowser/MainWindow.xaml.cs#L842-L878) in the **WebView2WpfBrowser** sample.
+
+```csharp
+        void DOMContentLoadedCmdExecuted(object target, ExecutedRoutedEventArgs e)
+        {
+            // Subscribe to the FrameCreated event to obtain the frame object when 
+            // it's created.
+            webView.CoreWebView2.FrameCreated += WebView_FrameCreatedDOMContentLoaded;
+            webView.NavigateToString(@"<!DOCTYPE html>" +
+                                      "<h1>DOMContentLoaded sample page</h1>" +
+                                      "<h2>The content to the iframe and below will be added after DOM content is loaded </h2>" +
+                                      "<iframe style='height: 200px; width: 100%;'/>");
+        }
+
+        void WebView_FrameCreatedDOMContentLoaded(object sender, CoreWebView2FrameCreatedEventArgs args)
+        {
+            // In order for ExecuteScriptAsync to successfully run inside the iframe, 
+            // subscribe to the ContentLoading or DOMContentLoaded event.  Once these 
+            // events are raised, you can call ExecuteScriptAsync.
+            args.Frame.DOMContentLoaded += (frameSender, DOMContentLoadedArgs) =>
+            {
+                args.Frame.ExecuteScriptAsync(
+                    "let content = document.createElement(\"h2\");" +
+                    "content.style.color = 'blue';" +
+                    "content.textContent = \"This text was added to the iframe by the host app\";" +
+                    "document.body.appendChild(content);");
+            };
+        }
+```
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+<!-- todo -->
```

Comment:
Will this todo be completed before checking in? Or do we have task to complete it later?

Reply:
This todo is planned for after this PR is merged.

@vbryh-msft @nishitha-burman - Is there a work item to add WinRT code here that's equivalent to: 

.NET:
[MainWindow.xaml.cs](https://github.com/MicrosoftEdge/WebView2Samples/blob/main/SampleApps/WebView2WpfBrowser/MainWindow.xaml.cs#L842-L878) in the **WebView2WpfBrowser** sample.

Win32:
[ICoreWebView2Frame2::ExecuteScript method](https://learn.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2frame2#executescript)

Reply:
@david-risney - work item title: "[WinRT] method AddHostObjectToScriptWithOrigins for frame object"